src/map/script.c

```diff
@@ -8779,29 +8779,25 @@ static BUILDIN(getcharid)
  *------------------------------------------*/
 static BUILDIN(getnpcid)
 {
-	int num = script_getnum(st,2);
-	struct npc_data* nd = NULL;
+	struct npc_data *nd = NULL;
 
-	if( script_hasdata(st,3) )
-	{// unique npc name
-		if( ( nd = npc->name2id(script_getstr(st,3)) ) == NULL )
-		{
-			ShowError("buildin_getnpcid: No such NPC '%s'.\n", script_getstr(st,3));
-			script_pushint(st,0);
-			return false;
+	// unique npc name
+	if (script_hasdata(st, 2)) {
+		// Deprecate old form - getnpcid(<type>{, <"npc name">})
+		if (script_isinttype(st, 2)) {
+			ShowWarning("buildin_getnpcid: Use of type is deprecated. Format - getnpcid({<\"npc name\">})\n");
+			script_pushint(st, 0);
+			return true;
 		}
-	}
 
-	switch (num) {
-		case 0:
-			script_pushint(st,nd ? nd->bl.id : st->oid);
-			break;
-		default:
-			ShowError("buildin_getnpcid: invalid parameter (%d).\n", num);
-			script_pushint(st,0);
-			return false;
+		if ((nd = npc->name2id(script_getstr(st, 2))) == NULL) {
+			script_pushint(st, 0);
+			return true;
+		}
 	}
 
+	script_pushint(st, (nd != NULL) ? nd->bl.id : st->oid);
```

Comment:
script_pushint with condition better put inside previous if block and add return

and outside of if block use only this ``script_pushint(st, st->oid);`` because outside of if block nd is always NULL